server/src/business-layer/services/StripeService.ts

```diff
@@ -22,6 +36,87 @@ export default class StripeService {
     return result.data
   }
 
+  /**
+   *
+   * @param email ideally extracted from JWT
+   * @param uid ideally extracted from JWT
+   */
+  public async createNewUser(email: string, uid: string) {
+    return await stripe.customers.create({
+      email,
+      metadata: { [USER_ID_KEY]: uid }
+    })
+  }
+
+  /**
+   * Checks for processing payments from a user. Used to avoid
+   * users from double paying after completing a checkout session for membership payments.
+   *
+   * @warning This assumes that users won't be paying for membership right after booking
+   *
+   * @param customerId `stripe_id` from the firebase document
+   */
+  public async hasProcessingPaymentIntent(customerId: string) {
+    const { data } = await stripe.paymentIntents.list({
+      customer: customerId
+    })
+    const hasProcessingPaymentIntent = !!data.find(
+      (intent) => intent.status === "processing"
```

Comment:
I notice that the docs for [intent statuses](https://docs.stripe.com/payments/paymentintents/lifecycle#intent-statuses) mention that
> These types of payment methods can take up to a few days to process.

I wonder if it's even worth having this check? Probably won't harm us, but we might want to measure later on how often people hit a processing intent and decide if we can cut this out to simplify the code.

Reply:
Have created #287